### PR TITLE
BBMRI-ID Virtual attribute module

### PR DESCRIPTION
**Pull request by dBucik**
Attribute module to extract original BBMRI ID as an attribute value from the UserExtSource. Serves for the purposes of migrating BBMRI AAI into the LS AAI. The module requires naming the ExtSource "BBMRI id" and being of type "ExtSourceInternal".

This attribute will be used by the BBMRI ProxyIdP temporarily until it migrates to LS ID. When we do so, the attribute module can be deprecated and consequently removed.

The attribute also needs to be exposed in LDAP.

### CODE REVIEW

File: perun-ldapc/src/main/resources/perun-ldapc.xml [hunk 1]
```diff
@@ -1273,6 +1273,16 @@ http://www.springframework.org/schema/aop http://www.springframework.org/schema/
 						</bean>
 					</property>
 				</bean>
+				<bean class="cz.metacentrum.perun.ldapc.model.impl.PerunAttributeDesc">
```
Comment (Johaney-s) [inline perun-ldapc/src/main/resources/perun-ldapc.xml:1276]: doesn't `perun-schema.ldif` also need to be updated?

File: perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_bbmriUserIdTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+			Arrays.asList(ues1, ues2, correctUes)
+		);
+
+		Attribute receivedAttr = classInstance.getAttributeValue(session, user, classInstance.getAttributeDefinition());
+		assertTrue(receivedAttr.getValue() instanceof Integer);
+		assertEquals("received attribute name is wrong",
+			classInstance.getAttributeDefinition().getFriendlyName(), receivedAttr.getFriendlyName());
+
+		@SuppressWarnings("unchecked")
+		Integer actual = (Integer) receivedAttr.getValue();
```
Comment (Johaney-s) [inline perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_bbmriUserIdTest.java:72]:
```suggestion
		Integer actual = receivedAttr.valueAsInteger();
```

Comment (dBucik) [reply perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_bbmriUserIdTest.java:72]: Fixed on all places.

File: perun-ldapc/src/main/resources/perun-ldapc.xml (hunk 1 above)
Comment (dBucik) [reply perun-ldapc/src/main/resources/perun-ldapc.xml:1276]: Please see the updated version.

**Comment by Johaney-s** [comment]
Solved in https://github.com/CESNET/perun/pull/3943

**Pull request closed without merging**